utils/python/CIME/SystemTests/README

```diff
@@ -1,11 +1,13 @@
 The following are the test functionality categories:
   1) smoke tests
-  2) restart tests
-  3) threading/pe-count modification tests
-  4) sequencing (layout) modification tests
-  5) multi-instance tests
-  6) performance tests
-  7) spinup tests (TODO)
+  2) basic reproducibility tests
```

Comment:
@rljacob : @jedwards4b says config_tests.xml is being moved. I get the sense that there is some disagreement about where the documentation of the tests should live, so I'd rather not make a unilateral decision to remove the documentation from config_tests.xml in this PR.